### PR TITLE
Remove panics from `Selector` message handling.

### PR DESCRIPTION
**Pull request by metasim**

### CODE REVIEW

File: src/blocks/selector.rs [hunk 1]
```diff
@@ -14,6 +9,11 @@ use crate::runtime::StreamIo;
 use crate::runtime::StreamIoBuilder;
 use crate::runtime::TypedBlock;
 use crate::runtime::WorkIo;
+use anyhow::anyhow;
+use std::cmp;
+use std::fmt;
+use std::ptr;
+use std::str::FromStr;
```
Comment (metasim) [inline src/blocks/selector.rs:16]: `cargo +nightly fmt --check` required this change.

File: src/blocks/selector.rs [hunk 2]
```diff
@@ -108,6 +108,16 @@ where
         )
     }
 
+    fn pmt_to_index(p: &Pmt) -> Result<Option<usize>> {
+        match p {
+            Pmt::U32(v) => Ok(Some(*v as usize % N)),
+            Pmt::U64(v) => Ok(Some(*v as usize % N)),
+            Pmt::Usize(v) => Ok(Some(*v % N)),
+            Pmt::Finished | Pmt::Ok => Ok(None),
```
Comment (metasim) [inline src/blocks/selector.rs:116]: Is this the right way to handle `Pmt::Finished`?

File: src/blocks/selector.rs [hunk 3]
```diff
@@ -108,6 +108,16 @@ where
...
+            o => Err(anyhow!("Invalid index specification: {:?}", o).into()),
```
Comment (metasim) [inline src/blocks/selector.rs:117]: Is `anyhow!` ok here?

**Comment by bastibl** [comment]
Great, thank you!

**Pull request merged**